pages/archive.js

```diff
@@ -1,23 +1,46 @@
 import FlashcardList from "@/components/FlashcardList/FlashcardList";
 import styled from "styled-components";
+import FormFlashcard from "@/components/FormFlashcard/FormFlashcard";
 
 export default function Archive({
   flashcardsWithCollection,
   handleToggleCorrect,
+  collections,
+  currentFlashcard,
+  setCurrentFlashcard,
+  handleEditFlashcard,
   handleDelete,
+  isEdit,
+  setIsEdit,
+  handleCreateFlashcard,
 }) {
   const correctFlashcards = flashcardsWithCollection.filter(
     (flashcard) => flashcard.isCorrect === true
   );
 
   return (
     <>
+      {correctFlashcards.length > 0 && isEdit && (
+        <FormFlashcard
+          onCreateFlashcard={handleCreateFlashcard}
+          collections={collections}
+          headline="Edit Flashcard"
+          isEdit={isEdit}
+          setIsEdit={setIsEdit}
```

Comment:
⚠️ regarding the state setter function issue. 

I will not comment further on this in the following files.... 😁